packages/backend/src/queryBuilder.ts

```diff
@@ -94,25 +95,23 @@ const getMetricFromId = (
     return metric;
 };
 
-export const replaceUserAttributes = (
+const replaceAttributes = (
+    regex: RegExp,
     sqlFilter: string,
-    userAttributes: UserAttributeValueMap,
-    stringQuoteChar: string = "'",
-    filter: string = 'sql_filter',
+    userAttributes: Record<string, string | string[]>,
```

Comment:
It cannot just be `UserAttributeValueMap` because it is a `Record<string, string[]>;` - has `string` arrays as the value and the intrinsic user attributes type has just `string` as the value

Also the reason why it cannot be `UserAttributeValueMap | IntrinsicUserAttributes` is because `IntrinsicUserAttributes` has exact object keys -`email`, `first_name`, `last_name` and ts wouldn't let me pass a string as the key to index the object, and so I wouldn't be able to reuse the function for both types like we are doing here

![image](https://github.com/lightdash/lightdash/assets/22939015/2ff968a1-f7c2-4e93-8779-0dd52e3db0f4)

I can change this to be `UserAttributeValueMap | Record<string, string>` but I thought this way it looked cleaner